src/Ball2.jl

```diff
@@ -209,3 +209,43 @@ function ⊆(B::Ball2{N}, S::AbstractSingleton{N}, witness::Bool=false
     end
     return (false, p)
 end
+
+"""
+    ⊆(B1::Ball2{N}, B2::Ball2{N}, witness::Bool=false
+     )::Union{Bool,Tuple{Bool,Vector{N}}} where {N<:Real}
+
+Check whether a ball in the 2-norm is contained in another ball in the 2-norm,
+and if not, optionally compute a witness.
+
+### Input
+
+- `B1` -- inner ball in the 2-norm
+- `B2` -- outer ball in the 2-norm
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``B1 ⊆ B2``
+* If `witness` option is activated:
+  * `(true, [])` iff ``B1 ⊆ B2``
+  * `(false, v)` iff ``B1 \\not\\subseteq B2`` and ``v ∈ B1 \\setminus B2``
+
+### Algorithm
+
+``B1 ⊆ B2`` iff ``‖ c_1 - c_2 ‖_2 + r_1 ≤ r_2``
+"""
+function ⊆(B1::Ball2{N}, B2::Ball2{N}, witness::Bool=false
+          )::Union{Bool,Tuple{Bool,Vector{N}}} where {N<:Real}
+    result = norm(B1.center - B2.center, 2) + B1.radius <= B2.radius
+    if witness
+        if result
+            return (result, N[])
+        end
+    else
+        return result
+    end
+
+    # compute a witness 'v'
+    v = B1.center .+ B1.radius * (B1.center .- B2.center)
+    return (false, v)
+end
```

Comment:
this would be in shortened form

```julia
v = result ? N[] : B1.center .+ B1.radius * (B1.center .- B2.center)
return witness ? (result, v) : result
```

although it has the drawback of computing `v` also when this is not really asked for.

Reply:
Yes, I definitely want to avoid this because I think in usual applications you do not want a witness.